plugin.py

```diff
@@ -224,7 +224,24 @@ def on_pre_send_notification_async(self, notification: Notification) -> None:
     # --- StoreListener ------------------------------------------------------------------------------------------------
 
     def on_store_changed(self, schemas: List[Dict]) -> None:
-        self._api.send_notification('json/schemaAssociations', [schemas + self._user_schemas])
+        self._api.send_notification('json/schemaAssociations', [self._unique_fileMatch(schemas + self._user_schemas)])
+
+    def _unique_fileMatch(self, schemas: List[Dict]) -> List[Dict]:
+        fileMatch = []  # type: List[Dict]
+        schemas_dict = {}  # type: Dict[str, str]
+        schemas_list = {}  # type: Dict[str, Dict]
+        for s in schemas:
+            if 'fileMatch' in s and isinstance(s['fileMatch'], list):
+                for f in s['fileMatch']:
+                    schemas_dict[f] = s['uri']
+            else:
+                fileMatch.append(s)
+        for match, uri in schemas_dict.items():
+            if match in schemas_list:
```

Comment:
The `schema_list` is keyed by `uri` so I don't think this check will ever by true.

Reply:
My local tests are working

Reply:
The code is wrong and will override the previous entry with the same URI.

So for existing schema like:

```
  {
    "fileMatch": [
      "/.cspell.json",
      "/cspell.json",
      "/cSpell.json"
    ],
    "uri": "https://raw.githubusercontent.com/streetsidesoftware/cspell/cspell4/cspell.schema.json"
  },
```

the resulting schema will only include one of the file matches.

Reply:
> The code is wrong and will override the previous entry with the same URI.

![image](https://user-images.githubusercontent.com/49985266/130743182-27abb845-3a0e-4581-ae56-7cb138dddfa9.png)

It is not overwritten, but appended to the fileMatch

Reply:
No because `if match in schemas_by_uri` is never gonna be true since the `schemas_by_uri` keys are "uris", not "file_matches".

Reply:
Thanks, I fix it